src/pages/workspace/withPolicyConnections.tsx

```diff
@@ -55,7 +55,7 @@ function withPolicyConnections<TProps extends WithPolicyConnectionsProps>(Wrappe
             openPolicyAccountingPage(props.policy.id);
         }, [props.policy?.id, isConnectionDataFetchNeeded]);
 
-        if ((isConnectionDataFetchNeeded || isFetchingData) && shouldBlockView) {
+        if ((isConnectionDataFetchNeeded || isFetchingData || isOnyxDataLoading) && shouldBlockView) {
```

Comment:
There will be a loading indicator if the onyx data isn't loaded yet, so it will show the first time you open the page. (or I believe any page that uses this HOC)